include/fprotect.h

```diff
@@ -55,6 +55,21 @@ int fprotect_area(uint32_t start, size_t length);
 int fprotect_area_no_access(uint32_t start, size_t length);
 #endif
 
+#ifdef CONFIG_FPROTECT_ENABLE_IS_PROTECTED
+/**
+ * @brief Check whether a block has already been protected.
+ *
+ * @param[in]  addr  The address to check. The block containing this address
+ *                   will be checked.
+ *
+ * @retval 0   If not protected
+ * @retval 1   If only write protected
+ * @retval 2   If only read protected
+ * @retval 3   If write and read protected
+ */
+uint32_t fprotect_is_protected(uint32_t addr);
```

Comment:
My bad, but according to the code standard one should not ifdef declarations, only definitions. This to avoid compilation errors when using `if (IS_ENABLED(...))`.

Could you move this ifdef out to the function definition in the .c file?

Reply:
thanks, done